hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/map/SpecificPartitionsImapReaderPmsTest.java

```diff
@@ -22,110 +22,233 @@
-        initialize(5, null);
+        initialize(MEMBERS, null);
     }
 
     @Before
     public void before() throws Exception {
+        partitionAssignment = getPartitionAssignment(instance());
+        reversedPartitionAssignment = new HashMap<>();
+        for (Map.Entry<Address, int[]> entry : partitionAssignment.entrySet()) {
+            for (int pId : entry.getValue()) {
+                reversedPartitionAssignment.put(pId, entry.getKey());
+            }
+        }
+
         mapName = randomName();
         sinkName = randomName();
+        sourceMap = instance().getMap(mapName);
+        sinkMap = instance().getMap(sinkName);
 
+        // By this, we ensure that PMS works without member pruning being used.
         jobConfig = new JobConfig().setArgument(SQL_ARGUMENTS_KEY_NAME, emptyList());
 
-        Map<Address, int[]> partitionAssignment = getPartitionAssignment(instance());
-        Address ownderAddress = instance().getCluster().getLocalMember().getAddress();
+        JobCoordinationService jobCoordinationService = ((JetServiceBackend) getNodeEngineImpl(instance())
+                .getService(JetServiceBackend.SERVICE_NAME))
+                .getJobCoordinationService();
+
+        jobInvocationObserver = new JobInvocationObserverImpl();
+        jobCoordinationService.registerInvocationObserver(jobInvocationObserver);
+
+        perMemberOwnedPKey = new int[MEMBERS - 1];
+        perMemberOwnedPId = new int[MEMBERS - 1];
+
+        Address coordinatorAddress = instance().getCluster().getLocalMember().getAddress();
         for (int i = 1; i < ITERATIONS; ++i) {
             int pIdCandidate = instance().getPartitionService().getPartition(i).getPartitionId();
-            if (Arrays.binarySearch(partitionAssignment.get(ownderAddress), pIdCandidate) >= 0) {
-                pKey = i;
-                keyOwnerPartitionId = pIdCandidate;
+            if (reversedPartitionAssignment.get(pIdCandidate).equals(coordinatorAddress)) {
+                coordinatorOwnedPartitionKey = i;
+                coordinatorOwnedPartitionId = pIdCandidate;
                 break;
             }
         }
+
+        Set<Address> clusterAddresses = new HashSet<>(partitionAssignment.keySet());
+        clusterAddresses.remove(coordinatorAddress);
+        int j = 0;
+        for (int i = 1; i < ITERATIONS && !clusterAddresses.isEmpty(); ++i) {
+            int pIdCandidate = getNodeEngineImpl(instance()).getPartitionService().getPartitionId(i);
+            Address address = reversedPartitionAssignment.get(pIdCandidate);
+            if (clusterAddresses.contains(address)) {
+                perMemberOwnedPKey[j] = i;
+                perMemberOwnedPId[j++] = pIdCandidate;
+                clusterAddresses.remove(address);
+            }
+        }
     }
 
     // Note: basic functionality must work even without member pruning used.
     @Test
     public void test_basic() {
         // Basic test is performed as submitting job by DAG.
```

Comment:
what is basic test supposed to test? before it used scan pruning, now it does not. Rename the method to be less confusing.